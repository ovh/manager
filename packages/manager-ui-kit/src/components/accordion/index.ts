export { Accordion } from './Accordion.component';
export type { AccordionProps } from './Accordion.props';

export { AccordionContent } from './accordion-content/AccordionContent.component';
export { AccordionItem } from './accordion-item/AccordionItem.component';
export { AccordionTrigger } from './accordion-trigger/AccordionTrigger.component';
