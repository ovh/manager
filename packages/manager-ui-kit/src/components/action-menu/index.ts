export { ActionMenu } from './ActionMenu.component';
export type { ActionMenuItem, ActionMenuProps } from './ActionMenu.props';
