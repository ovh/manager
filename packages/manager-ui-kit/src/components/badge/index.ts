export { Badge } from './badge.component';

export type { BadgeProps } from './badge.props';

export * from './Badge.constants';
