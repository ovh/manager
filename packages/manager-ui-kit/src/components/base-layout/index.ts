export { BaseLayout } from './BaseLayout.component';

export type { BaseLayoutProps } from './BaseLayout.props';

export type { HeaderProps } from './header/Header.props';
