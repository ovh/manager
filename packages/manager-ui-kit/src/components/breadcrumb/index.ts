export { Breadcrumb } from './Breadcrumb.component';
export type { BreadcrumbProps } from './Breadcrumb.props';
export * from './Breadcrumb.constants';
