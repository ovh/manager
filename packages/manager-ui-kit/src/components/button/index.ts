export { Button } from './Button.component';
export type { ButtonProps } from './Button.props';
