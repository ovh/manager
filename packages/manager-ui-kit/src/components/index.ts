export * from './action-banner';
export * from './action-menu';
export * from './redirection-guard';
export * from './breadcrumb';
export * from './base-layout';
export * from './clipboard';
export * from './step';
export * from './tabs';
export * from './tiles-input-group';
export * from './tiles-input';

export * from './Link';
export * from './drawer';
export * from './guide-menu';

export * from './notifications';

export * from './price';

export * from './filters';

export * from './button';
export * from './text';

export * from './service-state-badge';

export * from './tile';
export * from './order';

export * from './badge';
export * from './modal';
export * from './tags-list';
export * from './tags-tile';
export * from './changelog-menu';
export * from './link-card';
export * from './error';
export * from './error-boundary';
export * from './update-name-modal';
export * from './delete-modal';
export * from './grid-layout';
export * from './onboarding-layout';
export * from './datagrid';
export * from './accordion';
export * from './checkbox';
export * from './combobox';
export * from './datepicker';
export * from './file-upload';
export * from './form-field';
export * from './message';
export * from './phone-number';
export * from './popover';
export * from './progress';
export * from './quantity';
export * from './radio-group';
export * from './select';
export * from './switch';
export * from './timepicker';
export * from './toggle';
export * from './tooltip';
export * from './tree-view';
