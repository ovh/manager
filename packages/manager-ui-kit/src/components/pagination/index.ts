export { Pagination } from './Pagination.component';
export type { PaginationProps } from './Pagination.props';

export * from './constants/PaginationConstants';
