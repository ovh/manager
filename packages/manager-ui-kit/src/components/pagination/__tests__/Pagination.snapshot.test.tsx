import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import { Pagination } from '@/components';

describe('Pagination Snapshot tests', () => {
  it('renders the component with default props', () => {
    const { container } = render(
      <>
        <Pagination pageSize={25} totalItems={500} />
        <Pagination defaultPage={5} siblingCount={2} totalItems={500} />
        <Pagination
          labelTooltipNext="Go to next page"
          labelTooltipPrev="Go to previous page"
          totalItems={500}
        />
        <Pagination
          renderTotalItemsLabel={({ totalItems }) => `of ${totalItems} results`}
          totalItems={500}
          withPageSizeSelector
        />
      </>,
    );
    expect(container).toMatchSnapshot();
  });
});
