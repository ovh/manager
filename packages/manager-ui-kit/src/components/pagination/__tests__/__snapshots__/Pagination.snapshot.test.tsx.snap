// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pagination Snapshot tests > renders the component with default props 1`] = `
<div>
  <nav
    aria-label="pagination"
    class="_pagination_6qm34_2"
    data-ods="pagination"
    data-part="root"
    data-scope="pagination"
    dir="ltr"
    id="pagination::r0:"
  >
    <button
      aria-label="previous page"
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151"
      data-disabled=""
      data-ods="button"
      data-part="prev-trigger"
      data-scope="pagination"
      dir="ltr"
      disabled=""
      id="pagination::r0::prev"
      type="button"
    >
      <span
        class="_icon_10c23_2 _icon--chevron-left_10c23_88"
        data-ods="icon"
        role="presentation"
      />
    </button>
    <button
      aria-current="page"
      aria-label="page 1"
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--default_1koyw_49 _pagination__item_i63de_1"
      data-index="1"
      data-ods="button"
      data-part="item"
      data-scope="pagination"
      data-selected=""
      dir="ltr"
      id="pagination::r0::item:1"
      type="button"
    >
      1
    </button>
    <button
      aria-label="page 2"
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151 _pagination__item_i63de_1"
      data-index="2"
      data-ods="button"
      data-part="item"
      data-scope="pagination"
      dir="ltr"
      id="pagination::r0::item:2"
      type="button"
    >
      2
    </button>
    <button
      aria-label="page 3"
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151 _pagination__item_i63de_1"
      data-index="3"
      data-ods="button"
      data-part="item"
      data-scope="pagination"
      dir="ltr"
      id="pagination::r0::item:3"
      type="button"
    >
      3
    </button>
    <button
      aria-label="page 4"
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151 _pagination__item_i63de_1"
      data-index="4"
      data-ods="button"
      data-part="item"
      data-scope="pagination"
      dir="ltr"
      id="pagination::r0::item:4"
      type="button"
    >
      4
    </button>
    <button
      aria-label="page 5"
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151 _pagination__item_i63de_1"
      data-index="5"
      data-ods="button"
      data-part="item"
      data-scope="pagination"
      dir="ltr"
      id="pagination::r0::item:5"
      type="button"
    >
      5
    </button>
    <button
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151"
      data-ods="button"
      data-part="ellipsis"
      data-scope="pagination"
      dir="ltr"
      disabled=""
      id="pagination::r0::ellipsis:5"
      type="button"
    >
      …
    </button>
    <button
      aria-label="last page, page 20"
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151 _pagination__item_i63de_1"
      data-index="20"
      data-ods="button"
      data-part="item"
      data-scope="pagination"
      dir="ltr"
      id="pagination::r0::item:20"
      type="button"
    >
      20
    </button>
    <button
      aria-label="next page"
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151"
      data-ods="button"
      data-part="next-trigger"
      data-scope="pagination"
      dir="ltr"
      id="pagination::r0::next"
      type="button"
    >
      <span
        class="_icon_10c23_2 _icon--chevron-right_10c23_97"
        data-ods="icon"
        role="presentation"
      />
    </button>
  </nav>
  <nav
    aria-label="pagination"
    class="_pagination_6qm34_2"
    data-ods="pagination"
    data-part="root"
    data-scope="pagination"
    dir="ltr"
    id="pagination::r1:"
  >
    <button
      aria-label="previous page"
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151"
      data-ods="button"
      data-part="prev-trigger"
      data-scope="pagination"
      dir="ltr"
      id="pagination::r1::prev"
      type="button"
    >
      <span
        class="_icon_10c23_2 _icon--chevron-left_10c23_88"
        data-ods="icon"
        role="presentation"
      />
    </button>
    <button
      aria-label="page 1"
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151 _pagination__item_i63de_1"
      data-index="1"
      data-ods="button"
      data-part="item"
      data-scope="pagination"
      dir="ltr"
      id="pagination::r1::item:1"
      type="button"
    >
      1
    </button>
    <button
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151"
      data-ods="button"
      data-part="ellipsis"
      data-scope="pagination"
      dir="ltr"
      disabled=""
      id="pagination::r1::ellipsis:1"
      type="button"
    >
      …
    </button>
    <button
      aria-label="page 3"
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151 _pagination__item_i63de_1"
      data-index="3"
      data-ods="button"
      data-part="item"
      data-scope="pagination"
      dir="ltr"
      id="pagination::r1::item:3"
      type="button"
    >
      3
    </button>
    <button
      aria-label="page 4"
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151 _pagination__item_i63de_1"
      data-index="4"
      data-ods="button"
      data-part="item"
      data-scope="pagination"
      dir="ltr"
      id="pagination::r1::item:4"
      type="button"
    >
      4
    </button>
    <button
      aria-current="page"
      aria-label="page 5"
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--default_1koyw_49 _pagination__item_i63de_1"
      data-index="5"
      data-ods="button"
      data-part="item"
      data-scope="pagination"
      data-selected=""
      dir="ltr"
      id="pagination::r1::item:5"
      type="button"
    >
      5
    </button>
    <button
      aria-label="page 6"
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151 _pagination__item_i63de_1"
      data-index="6"
      data-ods="button"
      data-part="item"
      data-scope="pagination"
      dir="ltr"
      id="pagination::r1::item:6"
      type="button"
    >
      6
    </button>
    <button
      aria-label="page 7"
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151 _pagination__item_i63de_1"
      data-index="7"
      data-ods="button"
      data-part="item"
      data-scope="pagination"
      dir="ltr"
      id="pagination::r1::item:7"
      type="button"
    >
      7
    </button>
    <button
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151"
      data-ods="button"
      data-part="ellipsis"
      data-scope="pagination"
      dir="ltr"
      disabled=""
      id="pagination::r1::ellipsis:7"
      type="button"
    >
      …
    </button>
    <button
      aria-label="last page, page 50"
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151 _pagination__item_i63de_1"
      data-index="50"
      data-ods="button"
      data-part="item"
      data-scope="pagination"
      dir="ltr"
      id="pagination::r1::item:50"
      type="button"
    >
      50
    </button>
    <button
      aria-label="next page"
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151"
      data-ods="button"
      data-part="next-trigger"
      data-scope="pagination"
      dir="ltr"
      id="pagination::r1::next"
      type="button"
    >
      <span
        class="_icon_10c23_2 _icon--chevron-right_10c23_97"
        data-ods="icon"
        role="presentation"
      />
    </button>
  </nav>
  <nav
    aria-label="pagination"
    class="_pagination_6qm34_2"
    data-ods="pagination"
    data-part="root"
    data-scope="pagination"
    dir="ltr"
    id="pagination::r2:"
  >
    <button
      aria-label="previous page"
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151 _tooltip-trigger_wxukn_2"
      data-disabled=""
      data-ods="tooltip-trigger"
      data-part="trigger"
      data-scope="tooltip"
      data-state="closed"
      dir="ltr"
      disabled=""
      id="tooltip::r3::trigger"
      role="button"
      tabindex="0"
      type="button"
    >
      <span
        class="_icon_10c23_2 _icon--chevron-left_10c23_88"
        data-ods="icon"
        role="presentation"
      />
    </button>
    <button
      aria-current="page"
      aria-label="page 1"
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--default_1koyw_49 _pagination__item_i63de_1"
      data-index="1"
      data-ods="button"
      data-part="item"
      data-scope="pagination"
      data-selected=""
      dir="ltr"
      id="pagination::r2::item:1"
      type="button"
    >
      1
    </button>
    <button
      aria-label="page 2"
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151 _pagination__item_i63de_1"
      data-index="2"
      data-ods="button"
      data-part="item"
      data-scope="pagination"
      dir="ltr"
      id="pagination::r2::item:2"
      type="button"
    >
      2
    </button>
    <button
      aria-label="page 3"
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151 _pagination__item_i63de_1"
      data-index="3"
      data-ods="button"
      data-part="item"
      data-scope="pagination"
      dir="ltr"
      id="pagination::r2::item:3"
      type="button"
    >
      3
    </button>
    <button
      aria-label="page 4"
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151 _pagination__item_i63de_1"
      data-index="4"
      data-ods="button"
      data-part="item"
      data-scope="pagination"
      dir="ltr"
      id="pagination::r2::item:4"
      type="button"
    >
      4
    </button>
    <button
      aria-label="page 5"
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151 _pagination__item_i63de_1"
      data-index="5"
      data-ods="button"
      data-part="item"
      data-scope="pagination"
      dir="ltr"
      id="pagination::r2::item:5"
      type="button"
    >
      5
    </button>
    <button
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151"
      data-ods="button"
      data-part="ellipsis"
      data-scope="pagination"
      dir="ltr"
      disabled=""
      id="pagination::r2::ellipsis:5"
      type="button"
    >
      …
    </button>
    <button
      aria-label="last page, page 50"
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151 _pagination__item_i63de_1"
      data-index="50"
      data-ods="button"
      data-part="item"
      data-scope="pagination"
      dir="ltr"
      id="pagination::r2::item:50"
      type="button"
    >
      50
    </button>
    <button
      aria-label="next page"
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151 _tooltip-trigger_wxukn_2"
      data-ods="tooltip-trigger"
      data-part="trigger"
      data-scope="tooltip"
      data-state="closed"
      dir="ltr"
      id="tooltip::r4::trigger"
      role="button"
      tabindex="0"
      type="button"
    >
      <span
        class="_icon_10c23_2 _icon--chevron-right_10c23_97"
        data-ods="icon"
        role="presentation"
      />
    </button>
  </nav>
  <nav
    aria-label="pagination"
    class="_pagination_6qm34_2"
    data-ods="pagination"
    data-part="root"
    data-scope="pagination"
    dir="ltr"
    id="pagination::r5:"
  >
    <div
      class="_pagination-page-size-selector_1cj93_2"
    >
      <div
        data-ods="select"
        data-part="root"
        data-scope="select"
        dir="ltr"
        id="select::r7:"
      >
        <div
          data-part="control"
          data-scope="select"
          data-state="closed"
          dir="ltr"
          id="select::r7::control"
        >
          <button
            aria-controls="select::r7::content"
            aria-disabled="false"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-invalid="false"
            aria-labelledby=":r6:"
            class="_select-control_1mzzf_2"
            data-empty="false"
            data-ods="select-control"
            data-part="trigger"
            data-scope="select"
            data-state="closed"
            dir="ltr"
            id="select::r7::trigger"
            role="combobox"
            type="button"
          >
            <span
              data-part="value-text"
              data-scope="select"
              dir="ltr"
            >
              10
            </span>
            <span
              aria-hidden="true"
              class="_icon_10c23_2 _icon--chevron-down_10c23_85 _select-control__icon_1mzzf_55"
              data-ods="icon"
              data-part="indicator"
              data-scope="select"
              data-state="closed"
              dir="ltr"
              role="presentation"
            />
          </button>
          <select
            aria-hidden="true"
            aria-labelledby="select::r7::label"
            id="select::r7::select"
            style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap; word-wrap: normal;"
            tabindex="-1"
          >
            <option
              selected=""
              value="10"
            >
              10
            </option>
            <option
              value="25"
            >
              25
            </option>
            <option
              value="50"
            >
              50
            </option>
            <option
              value="100"
            >
              100
            </option>
            <option
              value="300"
            >
              300
            </option>
          </select>
        </div>
      </div>
      <p
        class="_pagination-page-size-selector__label_1cj93_7"
        id=":r6:"
      >
        of 500 results
      </p>
    </div>
    <button
      aria-label="previous page"
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151"
      data-disabled=""
      data-ods="button"
      data-part="prev-trigger"
      data-scope="pagination"
      dir="ltr"
      disabled=""
      id="pagination::r5::prev"
      type="button"
    >
      <span
        class="_icon_10c23_2 _icon--chevron-left_10c23_88"
        data-ods="icon"
        role="presentation"
      />
    </button>
    <button
      aria-current="page"
      aria-label="page 1"
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--default_1koyw_49 _pagination__item_i63de_1"
      data-index="1"
      data-ods="button"
      data-part="item"
      data-scope="pagination"
      data-selected=""
      dir="ltr"
      id="pagination::r5::item:1"
      type="button"
    >
      1
    </button>
    <button
      aria-label="page 2"
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151 _pagination__item_i63de_1"
      data-index="2"
      data-ods="button"
      data-part="item"
      data-scope="pagination"
      dir="ltr"
      id="pagination::r5::item:2"
      type="button"
    >
      2
    </button>
    <button
      aria-label="page 3"
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151 _pagination__item_i63de_1"
      data-index="3"
      data-ods="button"
      data-part="item"
      data-scope="pagination"
      dir="ltr"
      id="pagination::r5::item:3"
      type="button"
    >
      3
    </button>
    <button
      aria-label="page 4"
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151 _pagination__item_i63de_1"
      data-index="4"
      data-ods="button"
      data-part="item"
      data-scope="pagination"
      dir="ltr"
      id="pagination::r5::item:4"
      type="button"
    >
      4
    </button>
    <button
      aria-label="page 5"
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151 _pagination__item_i63de_1"
      data-index="5"
      data-ods="button"
      data-part="item"
      data-scope="pagination"
      dir="ltr"
      id="pagination::r5::item:5"
      type="button"
    >
      5
    </button>
    <button
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151"
      data-ods="button"
      data-part="ellipsis"
      data-scope="pagination"
      dir="ltr"
      disabled=""
      id="pagination::r5::ellipsis:5"
      type="button"
    >
      …
    </button>
    <button
      aria-label="last page, page 50"
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151 _pagination__item_i63de_1"
      data-index="50"
      data-ods="button"
      data-part="item"
      data-scope="pagination"
      dir="ltr"
      id="pagination::r5::item:50"
      type="button"
    >
      50
    </button>
    <button
      aria-label="next page"
      class="_button_1koyw_2 _button--primary_1koyw_100 _button--md_1koyw_42 _button--ghost_1koyw_151"
      data-ods="button"
      data-part="next-trigger"
      data-scope="pagination"
      dir="ltr"
      id="pagination::r5::next"
      type="button"
    >
      <span
        class="_icon_10c23_2 _icon--chevron-right_10c23_97"
        data-ods="icon"
        role="presentation"
      />
    </button>
  </nav>
</div>
`;
